ntp-proto/src/packet/v5.rs

```diff
@@ -87,17 +87,10 @@ impl NtpFlags {
     }
 
     fn as_bits(&self) -> [u8; 2] {
-        let mut flags: u16 = 0;
+        let mut flags = self.unknown_leap as u8;
+        flags |= 0x02 * self.interleaved_mode as u8;
 
-        if self.unknown_leap {
-            flags |= 0x01;
-        }
-
-        if self.interleaved_mode {
-            flags |= 0x02;
-        }
-
-        flags.to_be_bytes()
+        [0x00, flags]
     }
```

Comment:
This does work, but I think the `if` version is more explicit. And I'm pretty sure the compiler will optimize both roughly the same.

Reply:
it doesn't, weirdly 

https://godbolt.org/z/x19779b87

I'm confused by that choice. I still think the version with the if is clearer.

Reply:
Yeah that is weird. With the full function it gets a bit better but definitely not the same: https://godbolt.org/z/3dcs8d3T1
I still think we should use the `if` for better readability. I will ask @marlonbaeten later what he thinks about it.